src/py/flwr/cli/new/templates/app/code/client.xgboost.py.tpl

```diff
@@ -0,0 +1,161 @@
+        bst = bst_input[
+            bst_input.num_boosted_rounds()
+            - self.num_local_round : bst_input.num_boosted_rounds()
+        ]
+
+        return bst
+
+    def fit(self, ins: FitIns) -> FitRes:
+        global_round = int(ins.config["global_round"])
+        if global_round == 1:
+            # First round local training
+            bst = xgb.train(
+                self.params,
+                self.train_dmatrix,
+                num_boost_round=self.num_local_round,
+                evals=[(self.valid_dmatrix, "validate"), (self.train_dmatrix, "train")],
+            )
+        else:
+            bst = xgb.Booster(params=self.params)
+            global_model = bytearray(ins.parameters.tensors[0])
+
+            # Load global model into booster
+            bst.load_model(global_model)
+
+            # Local training
+            bst = self._local_boost(bst)
+
+        # Save model
+        local_model = bst.save_raw("json")
+        local_model_bytes = bytes(local_model)
+
+        return FitRes(
+            status=Status(
+                code=Code.OK,
+                message="OK",
+            ),
+            parameters=Parameters(tensor_type="", tensors=[local_model_bytes]),
+            num_examples=self.num_train,
+            metrics={},
+        )
+
+    def evaluate(self, ins: EvaluateIns) -> EvaluateRes:
+        # Load global model
+        bst = xgb.Booster(params=self.params)
+        para_b = bytearray(ins.parameters.tensors[0])
+        bst.load_model(para_b)
+
+        # Run evaluation
+        eval_results = bst.eval_set(
+            evals=[(self.valid_dmatrix, "valid")],
+            iteration=bst.num_boosted_rounds() - 1,
+        )
+        auc = round(float(eval_results.split("\t")[1].split(":")[1]), 4)
+
+        global_round = ins.config["global_round"]
+        log(INFO, f"AUC = {auc} at round {global_round}")
+
+        return EvaluateRes(
+            status=Status(
+                code=Code.OK,
+                message="OK",
+            ),
+            loss=0.0,
+            num_examples=self.num_val,
+            metrics={"AUC": auc},
+        )
+
+
+def client_fn(context: Context):
+    # Load model and data
+    partition_id = context.node_config["partition-id"]
+    num_partitions = context.node_config["num-partitions"]
+    train_dmatrix, valid_dmatrix, num_train, num_val = load_data(
+        partition_id, num_partitions
+    )
+
+    num_local_round = context.run_config["local-epochs"]
+    params = {
+        "objective": "binary:logistic",
+        "eta": context.run_config["lr"],  # Learning rate
+        "max_depth": context.run_config["max-depth"],
+        "eval_metric": "auc",
+        "nthread": context.run_config["nthread"],
+        "num_parallel_tree": context.run_config["num-parallel-tree"],
+        "subsample": context.run_config["subsample"],
+        "tree_method": context.run_config["tree-method"],
```

Comment:
I'm wondering if it would be better if we  pass the run config directly and make use of it in the constructor of the client (just to avoid having a long list of args)

Reply:
How about this? In `pyproject.toml`, we have:

```
[tool.flwr.app.config]
params.objective = "binary:logistic"
params.eta = 0.1  # Learning rate
params.max_depth = 8
params.eval_metric = "auc"
params.nthread = 16
params.num_parallel_tree = 1
params.subsample = 1
params.tree_method = "hist"
```

Then, in client_app.py:

```python
cfg = unflatten_dict(context.run_config)
params = cfg["params"]
```

Reply:
oh yes! a way superior alternative to what i had in mind.